src/containers/shared/components/TokenSearchResults/TokenSearchResults.tsx

```diff
@@ -0,0 +1,81 @@
+import { useContext, useState } from 'react'
+import './styles.scss'
+
+import { useTranslation } from 'react-i18next'
+import axios from 'axios'
+import { useQuery } from 'react-query'
+import { useAnalytics } from '../../analytics'
+import { TokenSearchRow } from './TokenSearchRow'
+import SocketContext from '../../SocketContext'
+import Log from '../../log'
+import { getAccountLines } from '../../../../rippled/lib/rippled'
+import { FETCH_INTERVAL_XRP_USD_ORACLE_MILLIS } from '../../utils'
+
+const ORACLE_ACCOUNT = 'rXUMMaPpZqPutoRszR29jtC8amWq3APkx'
+
+interface SearchResultsProps {
+  currentSearchValue: string
+  setCurrentSearchInput: (string) => void
+}
+
+const SearchResults = ({
+  currentSearchValue,
+  setCurrentSearchInput,
+}: SearchResultsProps): JSX.Element | null => {
+  const analytics = useAnalytics()
+  const { t } = useTranslation()
+  const rippledSocket = useContext(SocketContext)
+  const [tokens, setTokens] = useState<any[]>([])
+  const [XRPUSDPrice, setXRPUSDPrice] = useState(0.0)
+
+  useQuery(['fetchXRPToUSDRate'], () => fetchXRPToUSDRate(), {
+    refetchInterval: FETCH_INTERVAL_XRP_USD_ORACLE_MILLIS,
+  })
+
+  useQuery(['fetchTokens', currentSearchValue], () => fetchTokens())
```

Comment:
This should be using the output of these like this ` const {data: tokens} = useQuery(['fetchTokens', currentSearchValue], () => fetchTokens()).  Let useQuery handle states.

Reply:
with this, will we still be able to manually set the token state in other parts of the component? we have to manually clear out state when the user deletes what they've typed, otherwise a bug occurs where the search bar is empty but the results from the last entered value are still there

Reply:
nvm, i see what you mean here

Reply:
Fixed this to handle slow internet